packages/libp2p/src/libp2p.ts

```diff
@@ -98,8 +98,19 @@ export class Libp2pNode<T extends ServiceMap = Record<string, unknown>> extends
 
     components.events.addEventListener('peer:update', evt => {
       // if there was no peer previously in the peer store this is a new peer
-      if (evt.detail.previous == null) {
-        this.safeDispatchEvent('peer:discovery', { detail: evt.detail.peer })
+      if (evt.detail.previous === null) {
+
+        if(evt.detail.peer.addresses.length === 0) {
+          throw new CodeError('peer must have at least one multiaddr', codes.ERR_INVALID_MULTIADDR)
```

Comment:
This will cause an `uncaughtException` event to be emitted on the `process` object and node will crash.

If `peer.addresses` is empty it should just translate to `peerInfo.multiaddrs` being empty.

Reply:
Why would a peer emit an update event if it has no have no addresses though? Seems a bit counter-intuitive to a discover a peer with no dial-able addresses.

Reply:
`peer:update` happens when a peer's info is updated in the peer store.  This can be in response to any field, not just multiaddrs.

Regardless, the event callback invocation is not wrapped in a try/catch so we cannot throw here.

Reply:
I think the problem here is that the `detail.peer` field in `peer:update` is a [Peer](https://github.com/libp2p/js-libp2p/blob/master/packages/interface/src/peer-store/index.ts#L22-L53) instance, but the detail field of `peer:discovery` should be a [PeerInfo](https://github.com/libp2p/js-libp2p/blob/master/packages/interface/src/peer-info/index.ts#L4-L8) - `Peer` doesn't have a `.multiaddrs` field so the conversion in this PR is necessary for the emitted value to conform to the `PeerInfo` interface.